payjoin-cli/src/app/v2/mod.rs

```diff
@@ -104,26 +104,31 @@ impl AppTrait for App {
                 Ok(())
             }
             PjParam::V2(pj_param) => {
-                // TODO: perhaps we should store pj uri in the session wrapper as to not replay the event log for each session
+                let receiver_pubkey = pj_param.receiver_pubkey();
                 let sender_state =
                     self.db.get_send_session_ids()?.into_iter().find_map(|session_id| {
-                        let sender_persister =
-                            SenderPersister::from_id(self.db.clone(), session_id).ok()?;
-                        let (send_session, session_history) =
-                            replay_sender_event_log(&sender_persister)
+                        let session_receiver_pubkey = self
+                            .db
+                            .get_send_session_receiver_pk(&session_id)
+                            .expect("Receiver pubkey should exist if session id exists");
```

Comment:
This expect seems OK since it's looking for exactly one session to send. The other one seems less ok because the context does not restrict it to exactly one use.